pkg/server/helpers.go

```diff
@@ -206,8 +206,12 @@ func getRepoDigestAndTag(namedRef reference.Named, digest imagedigest.Digest, sc
 // localResolve resolves image reference locally and returns corresponding image metadata. It returns
 // nil without error if the reference doesn't exist.
 func (c *criContainerdService) localResolve(ctx context.Context, ref string) (*imagestore.Image, error) {
-	_, err := imagedigest.Parse(ref)
```

Comment:
not really convinced this is a good idea...

From my skim, sha256: should be a valid imagedigest.Parse(ref), and sha should not. Having a hard time thinking sha should be a valid imageStore.Get lookup. 

Thoughts?

Reply:
That said, if ```c.client.GetImage(ctx, normalized.String())``` below comes back with is not found... maybe we could do an attempt from c.imageStore.Get(normalized.String()) then check for error and return nil nil if is not found there either?

Reply:
Now we store sha256:xxx into imageStore .  so "sha" is a valid truncindex when patricia tree have only  one entry.  It is a bit of confused, but it is responsibility of the user. The user should give the truncID unambiguous.
@Random-Liu  What do you think?

Reply:
Yes, it's a valid truncindex reference, so long as there is only one image in the store, it is also an invalid reference. Won't we have two images in the store by default?  The pause image and one for it's container? Not sure if it's an interesting use case to support looking up sha to get the only image in the store.

Reply:
Let's put the test aside first. The pr mainly is for supporting truncindex in image.
I think that the image can not support truncindex if we have not the pr.
Do you give me a suggestion about supporting truncindex in image? 
@Random-Liu  I also need your suggestion .

Reply:
I don't think we need to add imagestore.Get above the parse test.. as a failure on parse calls get anyway. 

See below call to imageStore.Get on original lines 229-237.

Reply:
done